### PR TITLE
Improved testing for backend/app/mentorship

### LINKED ISSUE #2872: Improve backend/mentorship/* test coverage
**Describe the solution you'd like**
Add tests for better coverage of backend/mentorship/* -- then remove the coverage [exclusion](https://github.com/OWASP/Nest/blob/72341f0bc0c01c209c1f0ba7f1da110e73d016ba/backend/pyproject.toml#L156) 

**Are you going to work on implementing this?**

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by kart-u**
## Proposed change

Resolves #2872 

Added tests for app/mentorship 
Left out tests for `backend/apps/mentorship/api/internal/mutations` intentionally due some ongoing work on permissions -- #3101 #3185 (waiting for their completion)
coverage2

## Checklist

- [x] **Required:** I read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md)
- [x] **Required:** I ran `make check-test` locally and all tests passed
- [x] I used AI for code, documentation, or tests in this PR

### CODE REVIEW

File: backend/tests/apps/mentorship/model/test_module.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        mock_module.started_at = explicit_start
+        mock_module.ended_at = explicit_end
+        mock_create_module.return_value = mock_module
+
+        module = Module.objects.create(name="Auto Date Module", program=program_with_dates, project=self.project)
+        module.save()  
+
+        assert module.started_at == explicit_start
+        assert module.ended_at == explicit_end
+        assert module.key == "date-module"
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/model/test_module.py]: No need i just wanted to test if condition in save here which is being done

File: backend/tests/apps/mentorship/model/module_test.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            ended_at=datetime(2023, 12, 31, 18, 0, 0),
+        )
+        self.project = MagicMock(spec=Project, name="Test Project", key="test-project")
+
+    def test_str_returns_name(self):
+        """Unit test: __str__ should return the module name (pure mock)."""
+        mock_module = MagicMock(spec=Module)
+        mock_module.name = "Security Basics"
+
+        assert Module.__str__(mock_module) == "Security Basics"
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/model/module_test.py:30]: No need i just wanted a pure mock test

File: backend/tests/apps/mentorship/model/test_program.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class TestProgram:
+    
+    def test_program_status_choices(self):
+        assert Program.ProgramStatus.DRAFT == "draft"
+        assert Program.ProgramStatus.PUBLISHED == "published"
+        assert Program.ProgramStatus.COMPLETED == "completed"
+
+    def test_str_returns_name(self):
+        mock_program_instance = Program(name="Security Program")
+        assert Program.__str__(mock_program_instance) == "Security Program"
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/model/test_program.py]: No need i just wanted a pure mock test

File: backend/apps/mentorship/api/internal/nodes/mentee.py [hunk 4]
```diff
@@ -27,4 +27,4 @@ def resolve_avatar_url(self) -> str:
     @strawberry.field(name="experienceLevel")
     def resolve_experience_level(self) -> str:
         """Get the experience level of the mentee."""
-        return self.experience_level if self.experience_level else "beginner"
+        return self.experience_level if self.experience_level else "beginner"
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/apps/mentorship/api/internal/nodes/mentee.py]: file out of scope

[the remark this answers is not in the document]
Comment (kart-u) [reply backend/apps/mentorship/api/internal/nodes/mentee.py]: file out of scope

File: backend/tests/apps/mentorship/model/test_issue_user_interest.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+from unittest.mock import MagicMock, patch
+
+from apps.mentorship.models import IssueUserInterest, Module
+from apps.github.models import Issue, User
+
+
+class TestIssueUserInterest:
+    user = MagicMock(spec=User, login="testuser")
+    issue = MagicMock(spec=Issue, title="Test Issue")
+    module = MagicMock(spec=Module, name="Test Module")
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/model/test_issue_user_interest.py]: That file has been removed

File: backend/tests/apps/mentorship/api/internal/queries/api_queries_mentorship_test.py [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+from unittest.mock import MagicMock, patch
+
+import pytest
+from django.core.exceptions import ObjectDoesNotExist
+
+from apps.github.models import User as GithubUser
+from apps.mentorship.api.internal.nodes.mentee import MenteeNode
+from apps.mentorship.api.internal.queries.mentorship import MentorshipQuery
+from apps.mentorship.models import Mentee, MenteeModule, Module, Mentor
+
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/api/internal/queries/api_queries_mentorship_test.py]: files have been added check again

[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/api/internal/queries/api_queries_mentorship_test.py]: check now

**Comment by kart-u** [comment]
@coderabbitai generate docstrings

**Comment by kart-u** [comment]
@coderabbitai full review

File: backend/tests/apps/mentorship/api/internal/nodes/api_internal_module_test.py [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+            Task.objects.filter(
+                module=self,
+                issue__number=issue_number,
+                assigned_at__isnull=False,
+            )
+            .order_by("-assigned_at")
+            .values_list("assigned_at", flat=True)
+            .first()
+        )
+
```
[the remark this answers is not in the document]
Comment (kart-u) [reply backend/tests/apps/mentorship/api/internal/nodes/api_internal_module_test.py:83]: resolved in latest commit

**Comment by arkid15r** [comment]
@kart-u this is useful one. Is it ready for review?

**Comment by anurag2787** [comment]
Hi @kart-u i was going through the coverage report for the mentorship portal and found the following issues

No test file for
- `apps/mentorship/api/internal/mutations/module.py`
- `apps/mentorship/api/internal/mutations/program.py`

Have test coverage of less than 80
- `apps/mentorship/models/mixins/program.py`
- `apps/mentorship/models/module.py`

We should add more edge cases test to improve the coverage more than 80%

**Comment by kart-u** [comment]
> Hi @kart-u i was going through the coverage report for the mentorship portal and found the following issues
> 
> No test file for
> 
> * `apps/mentorship/api/internal/mutations/module.py`
> * `apps/mentorship/api/internal/mutations/program.py`
> 
> Have test coverage of less than 80
> 
> * `apps/mentorship/models/mixins/program.py`
> * `apps/mentorship/models/module.py`
> 
> We should add more edge cases test to improve the coverage more than 80%

I have already stated reason for no test file in `apps/mentorship/api/internal/mutations/module.py` and `apps/mentorship/api/internal/mutations/program.py` as there is some ongoing work, as for the edge cases I was planning to add them thats why this PR was draft earlier

**Comment by kart-u** [comment]
> @kart-u this is useful one. Is it ready for review?

Well I can do some more work on it as test for mutations are not added , But the tests added till now are ready for review
